packages/server/modules/core/repositories/users.ts

```diff
@@ -136,19 +140,28 @@ export async function getUserByEmail(
   email: string,
   options?: Partial<{ skipClean: boolean; withRole: boolean }>
 ) {
-  const q = Users.knex<UserWithOptionalRole[]>().whereRaw('lower(email) = lower(?)', [
-    email
-  ])
+  const userEmail = await db<UserEmail>(UserEmails.name)
```

Comment:
Why invoke this separate query? I'm sure its possible to have this function only invoke a single query to get the user by their email

Reply:
The first query is only used to check if the email exists.
To do that without the email field in the `users` table we should make the join with the `user_emails` table, which we do, but only if necessary.
If the email is not found we just return null with no other operations.

Reply:
@alemagio I mean, sure, but that doesn't answer my original question - why invoke this as a separate query when this could easily be one? Sure, the scenario when the email doesn't exist may be a simpler query, but then in all other scenarios the operation is twice as heavy as it needs to do 2 serial SQL queries, both the simple one and then a complex one right after.

Pretty much always it's preferable to use 1 query over 2, as long as the queries are relatively simple ones like the ones we have here